Libplanet/Blocks/BlockHeader.cs

```diff
@@ -337,11 +385,43 @@ internal void Validate(DateTimeOffset currentTime)
             if (!new HashDigest<SHA256>(PreEvaluationHash.ToArray()).Satisfies(Difficulty))
             {
                 throw new InvalidBlockNonceException(
-                    $"Block #{Index} {ByteUtil.Hex(Hash)}'s pre-evaluation hash " +
+                    $"Block #{Index} {hash}'s pre-evaluation hash " +
                     $"({ByteUtil.Hex(PreEvaluationHash)}) with the nonce " +
                     $"({ByteUtil.Hex(Nonce)}) does not satisfy its difficulty level {Difficulty}."
                 );
             }
+
+            HashDigest<SHA256> calculatedHash = Hashcash.Hash(
+                SerializeForHash(
+                    ProtocolVersion,
+                    Index,
+                    Timestamp,
+                    Difficulty,
+                    Nonce,
+                    Miner,
+                    PreviousHash,
+                    TxHash,
+                    StateRootHash
+                )
+            );
```

Comment:
It is a question, Could you describe what difference between the above code and `internal byte[] SerializeForHash()` code? It seems same to me.

```suggestion
            HashDigest<SHA256> calculatedHash = Hashcash.Hash(SerializeForHash());
```

Reply:
Yes, it's the same. 😅